[StarOS] INTERNAL_ERROR when stream_load transaction prepare

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
```
[hadoop@ip-172-31-46-234 StarRocksTest]$ nosetests-3.7 -s --with-xunit --xunit-file=./result/test_stream_load_trans_py_TestStreamLoadTransaction_test_load_begin_load_prepare_rollback.xml --verbose case/system_case/test_dml/test_stream_load_transaction/test_stream_load_trans.py:TestStreamLoadTransaction.test_load_begin_load_prepare_rollback --nologcapture
/usr/local/lib/python3.7/site-packages/nose/plugins/manager.py:395: RuntimeWarning: Unable to load plugin windmill = windmill.authoring.nose_plugin:WindmillNosePlugin: No module named 'bin'
  RuntimeWarning)
test_stream_load_trans.TestStreamLoadTransaction.test_load_begin_load_prepare_rollback ... {'host': '172.31.46.234', 'port': '9030', 'user': 'root', 'password': ''}
show backends;
create database stream_load_test_db_95707168_af4f_11ed_b3b0_069c711e01c6;
use stream_load_test_db_95707168_af4f_11ed_b3b0_069c711e01c6;
decline replication_num to 1 due to lack of backends, expected replication_num=3, actual be_num=2

 CREATE TABLE `duplicate_table_with_null` ( `k1`  date, `k2`  datetime, `k3`  char(20), `k4`  varchar(20), `k5`  boolean, `k6`  tinyint, `k7`  smallint, `k8`  int, `k9`  bigint, `k10` largeint, `k11` float, `k12` double, `k13` decimal(27,9) ) DUPLICATE KEY(`k1`, `k2`, `k3`, `k4`, `k5`) COMMENT "OLAP" DISTRIBUTED BY HASH(`k1`, `k2`, `k3`) BUCKETS 3 PROPERTIES ( "replication_num" = "1", "storage_format" = "v2" );
select count(1) from duplicate_table_with_null;
curl --location-trusted -u root: -H label:stream_load_trans_957d8d6c_af4f_11ed_b3b0_069c711e01c6 -H db:stream_load_test_db_95707168_af4f_11ed_b3b0_069c711e01c6 -H table:duplicate_table_with_null -XPOST http://172.31.46.234:8130/api/transaction/begin
{'Status': 'OK', 'Message': '', 'Label': 'stream_load_trans_957d8d6c_af4f_11ed_b3b0_069c711e01c6', 'TxnId': 23221, 'BeginTxnTimeMs': 0}
curl --location-trusted -u root: -H label:stream_load_trans_957d8d6c_af4f_11ed_b3b0_069c711e01c6 -H db:stream_load_test_db_95707168_af4f_11ed_b3b0_069c711e01c6 -H table:duplicate_table_with_null -T /mnt1/StarRocksTest/lib/../common/data/basic_types_data -XPUT http://172.31.46.234:8130/api/transaction/load
{'Status': 'OK', 'Message': '', 'Label': 'stream_load_trans_957d8d6c_af4f_11ed_b3b0_069c711e01c6', 'TxnId': 23221, 'LoadBytes': 8790855, 'StreamLoadPlanTimeMs': 0, 'ReceivedDataTimeMs': 2}
select count(1) from duplicate_table_with_null;
curl --location-trusted -u root: -H label:stream_load_trans_957d8d6c_af4f_11ed_b3b0_069c711e01c6 -H db:stream_load_test_db_95707168_af4f_11ed_b3b0_069c711e01c6 -XPOST http://172.31.46.234:8130/api/transaction/prepare
{'Status': 'INTERNAL_ERROR', 'Message': ''}
drop database stream_load_test_db_95707168_af4f_11ed_b3b0_069c711e01c6;
FAIL

======================================================================
FAIL: test_stream_load_trans.TestStreamLoadTransaction.test_load_begin_load_prepare_rollback
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/mnt1/StarRocksTest/case/system_case/test_dml/test_stream_load_transaction/test_stream_load_trans.py", line 282, in test_load_begin_load_prepare_rollback
    eq_(res["Status"], "OK", res)
AssertionError: {'Status': 'INTERNAL_ERROR', 'Message': ''}

----------------------------------------------------------------------
XML: /mnt1/StarRocksTest/result/test_stream_load_trans_py_TestStreamLoadTransaction_test_load_begin_load_prepare_rollback.xml
----------------------------------------------------------------------
Ran 1 test in 0.353s

FAILED (failures=1)
```

### Expected behavior (Required)

### Real behavior (Required)

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

--- a/fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java
+++ b/fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java
@@ -1135,7 +1135,7 @@ public class DatabaseTransactionMgr {
                     continue;
                 }
                 partitionCommitInfo.setVersion(partition.getNextVersion());
-                partitionCommitInfo.setVersionTime(commitTs);
+                partitionCommitInfo.setVersionTime(table.isLakeTable() ? 0 : commitTs);
             }
         }
 

--- a/fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnStateListener.java
+++ b/fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnStateListener.java
@@ -109,11 +109,15 @@ public class LakeTableTxnStateListener implements TransactionStateListener {
 
     @Override
     public void preWriteCommitLog(TransactionState txnState) {
-        Preconditions.checkState(txnState.getTransactionStatus() == TransactionStatus.COMMITTED);
+        Preconditions.checkState(txnState.getTransactionStatus() == TransactionStatus.COMMITTED
+                || txnState.getTransactionStatus() == TransactionStatus.PREPARED);
         TableCommitInfo tableCommitInfo = new TableCommitInfo(table.getId());
         for (long partitionId : dirtyPartitionSet) {
             Partition partition = table.getPartition(partitionId);
-            long version = partition.getNextVersion();
+            long version = -1;
+            if (txnState.getTransactionStatus() == TransactionStatus.COMMITTED) {
+                version = partition.getNextVersion();
+            }
             PartitionCommitInfo partitionCommitInfo = new PartitionCommitInfo(partitionId, version, 0);
             tableCommitInfo.addPartitionCommitInfo(partitionCommitInfo);
         }

